### PR TITLE
Harden REST client retry diagnostics

### LINKED ISSUE #20: M6-03 Hardening: retries, diagnostics, drift, and edge cases
Scope: harden retry classes, timeout behavior, diagnostics, drift messages, and pfSense API edge cases.

Acceptance:
- Transient and permanent errors are classified.
- Retry policy avoids duplicating unsafe writes.
- Mock tests cover common REST failure modes.

### PR DESCRIPTION
**Pull request by d3vi1**
## Summary

Closes #20.

- add REST client failure classification for transient vs permanent failures
- retry only safe `GET` requests once for transient HTTP/API-envelope/transport failures, honoring bounded `Retry-After`
- keep mutating `POST`/`PATCH`/`PUT`/`DELETE` requests non-replayed and add operator guidance for live pfSense state inspection
- document the read-retry/write-no-replay contract in `README.md`

## Validation

- `go test ./...`
- `make lint`
- `terraform fmt -check -recursive .`
- `git diff --check`

Fresh tester and adversarial review are running independently before merge.

### CODE REVIEW

File: internal/pfsense/client.go [hunk 1]
```diff
@@ -133,71 +185,209 @@ func (c *Client) Do(ctx context.Context, method string, path string, in any, out
...
+	if errors.As(err, &apiErr) {
+		return classifyStatusCode(apiErr.StatusCode)
+	}
+
+	if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+		return failureClassPermanent
+	}
+	var netErr net.Error
+	if errors.As(err, &netErr) && netErr.Timeout() {
+		return failureClassTransient
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply internal/pfsense/client.go]: Fixed in `22a069eb3919df0b371e1477b9ca845096d4ca86`: `classifyFailure` now checks `net.Error.Timeout()` before context deadline matching, and `TestGetRetriesClientTimeoutOnce` covers the real `http.Client.Timeout` retry path.

File: internal/pfsense/client.go [hunk 2]
```diff
@@ -133,71 +185,209 @@ func (c *Client) Do(ctx context.Context, method string, path string, in any, out
...
+	if err != nil {
+		return fmt.Errorf("read %s %s response: %w", method, displayPath, err)
+	}
+	if len(bytes.TrimSpace(responseBody)) == 0 {
+		return nil
+	}
+
+	if envelope, ok := parseEnvelope(responseBody); ok {
+		if envelope.Code >= 400 {
+			return envelopeAPIError(resp.StatusCode, method, displayPath, responseBody, envelope)
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply internal/pfsense/client.go]: Fixed in `22a069eb3919df0b371e1477b9ca845096d4ca86`: `Retry-After` is now passed into `envelopeAPIError`, stored on `APIError`, and covered by `TestEnvelopeRetryWaitHonorsRetryAfterAndContextCancellation`.
